hastory-data/src/Data/Hastory/Utils.hs

```diff
@@ -20,3 +31,19 @@ doCountsWith conv func = foldl go HM.empty
       where
         a Nothing = Just 1
         a (Just d) = Just $ d + func k
+
+dataBaseSpec :: SpecWith SqlBackend -> Spec
+dataBaseSpec = around withDatabase
+
+withDatabase :: ActionWith SqlBackend -> IO ()
+withDatabase = runNoLoggingT . withSqlConn setupDb . fmap liftIO
+
+setupDb :: LogFunc -> IO SqlBackend
```

Comment:
No need for this anymore then :)

Reply:
I think i still need `setupDb` in order to indicate how to setup the database - i.e., how to run the migration.

Reply:
Oh wauw now I'm even more confused.

I had something like this in mind:
https://github.com/NorfairKing/smos/blob/sha256/smos-sync-client-gen/src/Smos/Sync/Client/TestUtils.hs#L43-L48

Now I'm trying to understand what you've done differently. Something is definitely fishy.
Could it be that you're opening the connection twice?